lib/Models/GeoJsonCatalogItem.ts

```diff
@@ -0,0 +1,674 @@
+
+            return jsonPromise
+                .then((data: any) => {
+                    return this.setTraitsFromData(data);
+                })
+                .otherwise((e?: Error) => {
+                    if (e instanceof TerriaError) {
+                        throw e;
+                    } else {
+                        throw new TerriaError({
+                            sender: this,
+                            title: "Could not load JSON",
+                            message: `An error occurred while retrieving JSON data from the provided link.<p>If you entered the link manually, please verify that the link is correct.</p><p>This error may also indicate that the server does not support <a href="http://enable-cors.org/" target="_blank">CORS</a>. If this is your server, verify that CORS is enabled and enable it if it is not.  If you do not control the server, please contact the administrator of the server and ask them to enable CORS. Or, contact the ${appName} team by emailing <a href="mailto:${supportEmail}">${supportEmail}</a> and ask us to add this server to the list of non-CORS-supporting servers that may be proxied by ${appName} itself.</p><p>If you did not enter this link manually, this error may indicate that the data source you\'re trying to add is temporarily unavailable or there is a problem with your internet connection.  Try adding the data source again, and if the problem persists, please report it by sending an email to <a href="mailto:${supportEmail}">${supportEmail}</a>.</p>`
+                        });
+                    }
+                });
+        }
+    }
+
+    setTraitsFromData(geoJson: any): Promise<void> {
+        // If this GeoJSON data is an object literal with a single property, treat that
+        // property as the name of the data source, and the property's value as the
+        // actual GeoJSON.
+        var numProperties = 0;
+        var propertyName;
+        for (propertyName in geoJson) {
+            if (geoJson.hasOwnProperty(propertyName)) {
+                ++numProperties;
+                if (numProperties > 1) {
+                    break; // no need to count past 2 properties.
+                }
+            }
+        }
+
+        var name;
+        if (numProperties === 1 && isDefined(propertyName)) {
+            name = propertyName;
+            geoJson = geoJson[propertyName];
+            this.name = name;
+        }
+
+        // Reproject the features if they're not already EPSG:4326.
+        const proj4ServiceBaseUrl = this.catalogItem.terria.configParameters
+            .proj4ServiceBaseUrl;
+        return when(reprojectToGeographic(geoJson, proj4ServiceBaseUrl), () => {
+            this.geoJsonData = geoJson;
+        });
+    }
+}
+
+export default class GeoJsonCatalogItem
+    extends LoadGeoJsonMixin(
+        UrlMixin(CatalogMemberMixin(Model(GeoJsonCatalogItemTraits)))
+    )
+    implements Mappable {
+    @observable
+    show: boolean = true;
```

Comment:
This should be a trait instead of defined here.

Reply:
How about making it part of `CatalogMemberTraits`?

Reply:
Probably not `CatalogMemberTraits` because groups will have that, too, and it doesn't make sense to show/hide a group. Maybe a new `MappableTraits`?